src/lib/network/install_inf_convertor.rb

```diff
@@ -318,31 +314,31 @@ def create_wlan_ifcfg
       ifcfg = "WIRELESS_ESSID='#{wlan_essid}'\n"
 
       case wlan_auth
-        when "", "psk"
-          ifcfg << "WIRELESS_WPA_PSK='#{wlan_key}'\n"
-          ifcfg << "WIRELESS_AUTH_MODE='psk'\n"
-
-        when "open"
-          ifcfg << "WIRELESS_AUTH_MODE='no-encryption'\n"
-
-        when "wep_open", "wep_restricted"
-          if wlan_key_type == "password"
-            type = "h:"
-          elsif wlan_key_type == "ascii"
-            type = "s:"
-          else
-            type = wlan_key_type[0] + ":"
-          end
-
-          auth_mode = wlan_auth == "wep-open" ? "open" : "sharedkey"
-
-          ifcfg << "WIRELESS_AUTH_MODE='%s'\n" % wlan_auth_mode
-          ifcfg << "WIRELESS_DEFAULT_KEY='0'\n"
-          ifcfg << "WIRELESS_KEY_0='#{type}#{wlan_key}'\n"
-          ifcfg << "WIRELESS_KEY_LENGTH='#{wlan_key_len}'\n"
+      when "", "psk"
+        ifcfg << "WIRELESS_WPA_PSK='#{wlan_key}'\n"
+        ifcfg << "WIRELESS_AUTH_MODE='psk'\n"
+
+      when "open"
+        ifcfg << "WIRELESS_AUTH_MODE='no-encryption'\n"
+
+      when "wep_open", "wep_restricted"
+        if wlan_key_type == "password"
+          type = "h:"
+        elsif wlan_key_type == "ascii"
+          type = "s:"
+        else
+          type = wlan_key_type[0] + ":"
+        end
+
+        wlan_auth_mode = wlan_auth == "wep-open" ? "open" : "sharedkey"
```

Comment:
this is actually bug fix, as original auth_mode is not used and below wlan_auth_mode crash.